crates/topos-tce-transport/src/lib.rs

```diff
@@ -117,3 +134,62 @@ pub enum ProtocolEvents {
     /// Stable Sample
     StableSample,
 }
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq, Hash)]
+pub struct ValidatorId(H160);
+
+impl ValidatorId {
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+
+    pub fn address(&self) -> Address {
+        Address::from_str(&format!("0x{}", hex::encode(self.0))).unwrap()
+    }
+}
+
+impl From<H160> for ValidatorId {
+    fn from(address: H160) -> Self {
+        ValidatorId(address)
+    }
+}
+
+impl From<&str> for ValidatorId {
+    fn from(address: &str) -> Self {
+        ValidatorId(H160::from_str(address).expect("Cannot transform to H160"))
+    }
+}
+
+impl std::fmt::Display for ValidatorId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "0x{}", hex::encode(self.0))
+    }
+}
+
+pub async fn sign_message(
+    validator_id: ValidatorId,
+    certificate_id: CertificateId,
+    wallet: LocalWallet,
+) -> Result<Signature, WalletError> {
+    let mut hash = Vec::new();
+    hash.extend(certificate_id.as_array().iter().cloned());
+    hash.extend(validator_id.as_bytes());
+
+    let hash = keccak256(hash);
+
+    wallet.sign_message(hash.as_slice()).await
```

Comment:
This is going to hash the message, our `hash`, one more time. See https://docs.rs/ethers-core/2.0.10/src/ethers_core/utils/hash.rs.html#14